src/screens/CreatePost/index.js

```diff
@@ -473,8 +473,8 @@ const CreatePost = () => {
         }
 
         const post = await createPost(data);
-        if (params.onRefresh && typeof params.onRefresh === 'function') {
-          params.onRefresh();
+        if (params && params?.onRefresh && typeof params.onRefresh === 'function') {
+          params?.onRefresh();
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The use of optional chaining and nullish coalescing operators improves the safety and readability of the code. However, there's a redundancy in checking `params?.onRefresh` twice. You can simplify this by removing the first check.
```diff
- if (params && params?.onRefresh && typeof params.onRefresh === 'function') {
+ if (typeof params?.onRefresh === 'function') {
```
This will ensure that `params.onRefresh` is not undefined or null before checking its type. If `params` or `params.onRefresh` is undefined, it will not throw an error and the condition will simply evaluate to false.


<!-- This is an auto-generated comment by OSS CodeRabbit -->